core/storage/classifier/gae_models.py

```diff
@@ -174,3 +174,84 @@ def create(
 
         training_job_instance.put()
         return instance_id
+
+
+class ClassifierExplorationMappingModel(base_models.BaseModel):
+    """Model for mapping exploration attributes to a ClassifierDataModel.
+
+    The id of instances of this class has the form
+    {{exp_id}}.{{exp_version}}.{{state_name}}
+    """
+
+    # The ID of the classifier corresponding to the exploration attributes.
+    classifier_id = ndb.StringProperty(required=True, indexed=True)
+
+    @classmethod
+    def _generate_id(cls, exp_id, exp_version, state_name):
+        """Generates a unique ID for the Classifier Exploration Mapping of the
+        form {{exp_id}}.{{exp_version}}.{{state_name}}
+
+        Args:
+            exp_id: str. ID of the exploration.
+            exp_version: int. The exploration version at the time
+                this training job was created.
+            state_name: str. The name of the state to which the classifier
+                belongs.
+
+        Returns:
+            ID of the new Classifier Exploration Mapping instance.
+        """
+        if isinstance(state_name, unicode):
```

Comment:
I should clarify this: the reason I said "don't do it like this" was because this seems to introduce too much logic in the storage layer, which shouldn't really be responsible for handling this sort of thing in one-off ways. But if you just do something like utils.to_str(state_name) then I think that's fine.

Btw, did you figure out whether GAE IDs are represented as str or unicode?